### PR TITLE
Feature 2463 modify table

### LINKED ISSUE #2463: Documentation: Modify the format of the METplus Components Python Requirements Table
## Describe the Task ##
Modify the format of the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.  At the [METplus Engineering meeting on 20240109](https://docs.google.com/document/d/1SwFkyPvVdhUKvG6U-uui6Gty2fNkw5W4FldlI3sM9Vc/edit) (and at previous meetings) it was noted that this table is difficult to read because users have to scroll over and it is hard to keep track of the information in the row.

A drop down format was suggested in place of the table.  Mark up a few rows of the table (to start) in a feature branch to be in a drop down style format and get feedback from the team to see if this is a better way to proceed.  Please start by only doing a few rows and getting feedback from the team, before proceeding with converting the whole table. 

### Time Estimate ###
1 to 3 days of work.

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
None

### Funding Source ###
Per Tara, Lisa should split her time between 
2702691 | JNT DTC NSF BASE FUN…
2784543 | NOAA R2O METp SW2
2799991 | Met Office METplus S…

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
- [x] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issues
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by lisagoodrich**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
1.  The original section/table was listed in the overview as 1.5. METplus Components Python Requirements.  This section was REMOVED and added to the appendix, [12.2 METplus Components Python Packages] (https://metplus.readthedocs.io/en/feature_2463_modify_table/Users_Guide/appendixA.html#metplus-components-python-packages)
2.  The table has been swapped out for drop-down menus for readability. This was reviewed by @jprestop , @georgemccabe and @DanielAdriaansen to make sure this was an improvement over the table.  (It is an improvement.)
3. The "METplus Wrappers" language has been changed for clarity to "Select METplus Use Cases".  Except for the python3.10.4 +  and dateutil >= ` 2.8.2 drop-downs, where "METplus Wrappers" is the correct description.
4. Since the table was removed from the overview, a new label needed to be created to refer to the new dropdown menu section, 'components_python_packages'.  This is referred to in the add_use_case.rst file under the Contributor's Guide section.
5. To keep the version numbers up to date, two new entries were added to "rst_epilog" in the conf.py file: python_version and met_version

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Please review for any anomalies, such as wrong information, incorrect usage of "METplus Wrappers" or "Select METplus Use Cases."  

Here is the link: [12.2 METplus Components Python Packages] (https://metplus.readthedocs.io/en/feature_2463_modify_table/Users_Guide/appendixA.html#metplus-components-python-packages)

From discussing this with @georgemccabe at the METplus engineering meeting, some changes should be made to item 5 above.  George has a more elegant way to do this so that the versions can be updated in one place.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **[3/1/24]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Review by DanielAdriaansen — DISMISSED** [review]
Looks great! Thanks @lisagoodrich and @jprestop !

File: docs/Users_Guide/overview.rst
Comment (georgemccabe) [inline docs/Users_Guide/overview.rst]: Should `|python_version|` be used here instead of hard-coding 3.10? Also, line 73 noting deprecated support for 2.7 could be removed.

Comment (lisagoodrich) [reply docs/Users_Guide/overview.rst]: I think using |python_version| here would be a great idea.  (my 2 cents)

Comment (georgemccabe) [reply docs/Users_Guide/overview.rst]: Sounds good, @lisagoodrich. Can you make that change?

Comment (lisagoodrich) [reply docs/Users_Guide/overview.rst]: Thank you for your patience @georgemccabe. I have added "|python_version|" to line 72 and 
removed this: "We have deprecated support to Python 2.7."

Should updating  "|python_version|"  and  "|met_version|" be added to the Release Notes?  (If you'd like me to do this, please point me to the correct section.)

Are there any other changes you would like me to make?

Comment (georgemccabe) [reply docs/Users_Guide/overview.rst]: The Python version doesn't need to be updated for every release, so that isn't necessary to add to the release guide. It would be nice to automate the update of the MET version, but until that is done it should be updated as part of the release guide instructions. There isn't a clear place to put this in the release guide that doesn't involve moving things around, so it may be best to just create a GitHub issue to automate the update of the MET version in the documentation and assign it to me.

**Comment by lisagoodrich** [comment]
Okay, @georgemccabe, I did my best to create an issue to Automate the MET version number update #2509.  Please feel free to edit the issue.  I did my best but I don't always get the details right.

Is this issue ready to be squashed and merged?

**Comment by georgemccabe** [comment]
Thanks, @lisagoodrich. We changed the PR settings to require a re-review when changes are made after approval, so Dan or I will need to approve it again before we can merge.

We are testing out Slack reminders for open PRs, so I am going to hold off on approving to ensure that the reminder comes in tomorrow AM. Once I confirm that, I can review and merge this. Thanks again for working on this and creating the new issue.

**Comment by jprestop** [comment]
@georgemccabe I just wanted to follow up on this PR. You previously said:

> We are testing out Slack reminders for open PRs, so I am going to hold off on approving to ensure that the reminder comes in tomorrow AM. Once I confirm that, I can review and merge this. Thanks again for working on this and creating the new issue.

**Comment by georgemccabe** [comment]
Hi @jprestop, thanks for following up on this. I have been working on other projects lately, but I did send @JohnHalleyGotway a Slack message related to this:

> It triggered and send a reminder for Tina's PR, but it did not send one for Lisa's. It could be because Dan reviewed/approved it, but the PR is still blocked to merge until a re-review since we require a new review after new commits after a review.
> Sounds like feedback to provide to GitHub to improve it!

I can approve and merge this now.

**Review by georgemccabe — APPROVED** [review]
Looks good, thank you @lisagoodrich!

**Pull request merged**
